### DISCUSSION TITLE
Scale per clip

### DISCUSSION THREAD

**Discussion by steveotherr** [SynthstromAudible/DelugeFirmware#1367] [Ideas]
Being able to set a different scale per clip would be amazing for composing. If I want to modulate to a different key, use borrowed chords from outside the key, etc, I'm forced to use chromatic mode, which makes sequencing way more tedious/confusing than if I could just change the scale for each clip. Hope that makes sense and thank you for all of the amazing stuff you've done so far! :)

**Comment by dkrieger** [comment]
Rows in grid view are "sections", and they exist as audition pad colors in song row view and as clip colors in arranger view.

Also, how about independent scales per clip instead? More flexible and probably simpler to implement to allow clips to override the default song-level scale setting

**Reply by steveotherr** [reply]
Thank you for the correction and understanding what I meant. And yes, scales per clip would be just as good so if that's easier I'm all for it! I've edited the request/title to ask for scales per clip. :)

**Reply by DamianGolf666** [reply]
Yes please! This has became an issue for me as a performer. Not being able to switch both the root and scale type within a song, multiple times is hindering my performances. I know you can transpose the root note, and now you can also pick a scale type while playing in one of the newer keyboard modes, but trying to do this mid performance is a not feasible ie going from a verse to a bridge to a chorus. It needs to be a seamless process, as simple as change to a different coloured section.

**Comment by duncanrmi** [comment]
I'd like the scale to be switchable by the transposition track..... in fact, I'm inclining towards there being a master track which contains global data for the song- transpose, scale & maybe some mix automation.

**Comment by AtaDeluge** [comment]
Big **YES** on this one. With each clip having its own scale, it would make for much better composing and management of clips. Ideally this idea would also:

1. Indicate the scale on the screen when you hold down a clip on the song page (like how it shows the instrument)
2. Show the scale on the screen while on the clip page
3. Save and show the selected scale on one of the two right hand columns when on the clip page using a keyboard (if “Scale” was selected in one of those two right hand columns)

Scale management currently leaves a lot to be desired (as of 1.2) and this would be a significant improvement.

**Reply by AtaDeluge** [reply]
Hmm... It sort of seems like this has already been implemented, to a degree...

I was a little confused as to how the scales work, but I just tested it out and used two different scales on two different clips, and it remembered the scale when exiting scale mode and going back into scale mode and it even showed the scale on the right hand column in keyboard mode.

From the documentation 4.4.2:
"Improvement: when you exit Scale mode, and enter Scale mode again, if there was already a selected Scale, and it fits the notes you have entered, the Deluge will prefer that scale, instead of trying to change your scale or root note to other scale."

**Comment by thejesiah** [comment]
Kind of a workaround... with the original 7SEG Deluge, I've seen those users use individual projects as different sections of their songs. So naming is like 07A, 07B, 07C, etc. The Deluge does a really good job of not skipping a beat from one project to another... other than synth release, reverb tails, etc... 
It's a technique I think I need to implement anyway, because trying to keep track of 20 different sections using only 4 colors, in a single project just isn't working.
